eppo_client/client.py

```diff
@@ -183,114 +185,63 @@ def get_assignment_variation(
+            subject_attributes = {}
 
-        if experiment_config is None or not experiment_config.enabled:
-            logger.info(
-                "[Eppo SDK] No assigned variation. No active experiment or flag for key: "
-                + flag_key
-            )
+        flag = self.__config_requestor.get_configuration(flag_key)
+
+        if flag is None:
+            logger.info("[Eppo SDK] No assigned variation. Flag not found: " + flag_key)
             return None
 
-        matched_rule = find_matching_rule(subject_attributes, experiment_config.rules)
-        if matched_rule is None:
-            logger.info(
-                "[Eppo SDK] No assigned variation. Subject attributes do not match targeting rules: {0}".format(
-                    subject_attributes
-                )
+        if not check_type_match(expected_variation_type, flag.variation_type):
+            raise TypeError(
+                f"Variation value does not have the correct type."
+                f" Found: {flag.variation_type} != {expected_variation_type}"
             )
-            return None
 
-        allocation = experiment_config.allocations[matched_rule.allocation_key]
-        if not self._is_in_experiment_sample(
-            subject_key,
-            flag_key,
-            experiment_config.subject_shards,
-            allocation.percent_exposure,
-        ):
+        if not flag.enabled:
             logger.info(
-                "[Eppo SDK] No assigned variation. Subject is not part of experiment sample population"
+                "[Eppo SDK] No assigned variation. Flag is disabled: " + flag_key
             )
             return None
 
-        shard = get_shard(
-            "assignment-{}-{}".format(subject_key, flag_key),
-            experiment_config.subject_shards,
-        )
-        assigned_variation = next(
-            (
-                variation
-                for variation in allocation.variations
-                if is_in_shard_range(shard, variation.shard_range)
-            ),
-            None,
-        )
-
-        assigned_variation_value_to_log = None
-        if assigned_variation is not None:
-            assigned_variation_value_to_log = assigned_variation.value
-            if expected_variation_type is not None:
-                variation_is_expected_type = VariationType.is_expected_type(
-                    assigned_variation, expected_variation_type
-                )
-                if not variation_is_expected_type:
-                    return None
+        result = self.__evaluator.evaluate_flag(flag, subject_key, subject_attributes)
 
         assignment_event = {
-            "allocation": matched_rule.allocation_key,
-            "experiment": f"{flag_key}-{matched_rule.allocation_key}",
+            **(result.extra_logging if result else {}),
+            "allocation": result.allocation_key if result else None,
+            "experiment": f"{flag_key}-{result.allocation_key}" if result else None,
```

Comment:
What do we do with this? Should we revisit whether this is something worth logging? It's just a concatenation of the flag and the allocation, both of which are already logged.

Reply:
I think we should keep it as this is the column we filter on for experiment analysis. If we remove this, then users have to concatenate the other two columns